scattr/workflow/rules/mrtpipelines/filter_tck.smk

```diff
@@ -0,0 +1,261 @@
+            desc="{desc}",
+            suffix="mask.mif",
+        ),
+        **wildcards,
+        allow_missing=True,
+    )[0]
+    desc = glob_wildcards(exclude_mask).desc
+
+    return desc
+
+
+def get_filtered_tck(wildcards):
+    desc = get_desc(wildcards)
+    # Create params lists
+    return expand(
+        bids_tractography_out(
+            desc="{desc}",
+            suffix="tractography.tck",
+        ),
+        **wildcards,
+        desc=desc,
+        allow_missing=True,
+    )
+
+
+def get_filtered_weights(wildcards):
+    desc = get_desc(wildcards)
+    return expand(
+        bids_tractography_out(
+            desc="{desc}",
+            suffix="weights.csv",
+        ),
+        **wildcards,
+        desc=desc,
+        allow_missing=True,
+    )
+
+
+rule filter_combine_tck:
+    input:
+        unpack(aggregate_tck_files),
+        rules.connectome2tck.output.output_dir,
+        rules.create_exclude_mask.output.out_dir,
+        filter_mask=aggregate_exclude_masks,
+    params:
+        filtered_tck=get_filtered_tck,
+        filtered_weights=get_filtered_weights,
+        filtered_tck_exists=bids_tractography_out(
+            desc="from*", suffix="tractography.tck"
+        ),
+        filtered_weights_exists=bids_tractography_out(
+            desc="from*", suffix="weights.csv"
+        ),
+        exclude_mask_dir=bids_anat_out(
+            datatype="exclude_mask",
+        ),
+        unfiltered_tck_dir=str(
+            Path(
+                bids_tractography_out(
+                    datatype="unfiltered",
+                )
+            ).parent
+        ),
+    output:
+        combined_tck=bids_tractography_out(
+            desc="filteredsubcortical",
+            suffix="tractography.tck",
+        ),
+        combined_weights=bids_tractography_out(
+            desc="filteredsubcortical",
+            suffix="tckWeights.txt",
+        ),
+    threads: 1
+    resources:
+        tmp_dir=lambda wildcards: bids_tractography_out(
+            root=os.environ.get("SLURM_TMPDIR")
+            if config.get("slurm_tmpdir")
+            else "/tmp",
+            **wildcards,
+        ),
+        tmp_combined_tck=lambda wildcards: bids_tractography_out(
+            root=os.environ.get("SLURM_TMPDIR")
+            if config.get("slurm_tmpdir")
+            else "/tmp",
+            desc="filteredsubcortical",
+            suffix="tractography.tck",
+            **wildcards,
+        ),
+        tmp_combined_weights=lambda wildcards: bids_tractography_out(
+            root=os.environ.get("SLURM_TMPDIR")
+            if config.get("slurm_tmpdir")
+            else "/tmp",
+            desc="filteredsubcortical",
+            suffix="tckWeights.txt",
+            **wildcards,
+        ),
+        mem_mb=128000,
+        time=60,
+    log:
+        bids_log(suffix="combineFiltered.log"),
+    group:
+        "tractography_update"
+    container:
+        config["singularity"]["scattr"]
+    shell:
```

Comment:
I think this is getting long enough that it may be worth making it a script in an external file (debatable, though)

Reply:
Yeah, I don't mind doing that if we ended up adding more rules to this file - otherwise I'm happy to keep as is for now so that we don't have to navigate to another file to look at what the functions are doing.